assembly/index.ts

```diff
@@ -71,15 +72,61 @@ export function viewRandomDesign() : Design {
     logging.log(randomDesignIndex);
     
     let design : Design = designs.getSome(ownersValues[randomDesignIndex]);
-    // logging.log(`\n\n\t> Owner : ${design.owner} \n\n\t` + design.instructions.replaceAll("\n", "\n\t") + "\n")
+    // logging.log(`\n\n\t> Owner : ${design.owner_id} \n\n\t` + design.instructions.replaceAll("\n", "\n\t") + "\n")
 
     return design;
 }
 
+export function nft_token(token_id: string): Design | null { // token_id == owner_id
+    return designs.getSome(token_id);
+}
 
 function randomNum(max : u32 = <u32>context.blockIndex): u32 {
     const rng = new RNG<u32>(1, max);
     return rng.next()
 }
 
 
+export function nft_metadata(): NFTContractMetadata {
+    return new NFTContractMetadata();
+}
+
+
+export function nft_total_supply(): string {
+    return owners.size.toString();
+}
+
+
+export function nft_tokens(
+    from_index: string = "0",
```

Comment:
why not pass parameters as `u32` directly?

Reply:
It has to do with the fact that you might have a loss of precision (if you have enormous amounts of tokens) on the front-end side.
That's what I understood from reading the spec:
https://github.com/near/NEPs/blob/master/specs/Standards/NonFungibleToken/Enumeration.md